proposals/v3/api.md

```diff
@@ -0,0 +1,1119 @@
+| end | An RFC3339 or RFC3339Nano formatted timestamp which specifies an ending bound on the cache data to return. If no timestamp is specified, there will not be an ending bound. |
+
+#### Response
+
+##### OK (200)
+The response schema for the `/readcached` endpoint is the same as the response schema
+for the [`/read`](#read) endpoint.
+
+Unlike the [`/read`](#read) and [`/read/{device_id}`](#read-device) endpoints, the response for this endpoint is
+streamed JSON. One block of the streamed JSON will appear as follows:
+
+```json
+[
+  {
+    "device": "a72cs6519ee675b",
+    "kind": "temperature",
+    "type": "temperature",
+    "value": 20.3,
+    "timestamp": "2018-02-01T13:47:40.395939895Z",
+    "unit": {
+      "symbol": "C",
+      "name": "degrees celsius"
+    },
+    "context": {
+      "host": "127.0.0.1",
+      "sample_rate": 8
+    }
+  }
+]
+```
+
+**Fields**
+
+| Field | Description |
+| ----- | ----------- |
+| *device* | The ID of the device which the reading originated from. |
+| *kind* | The device kind, as specified by the plugin. |
+| *type* | The device type, as specified by the plugin. This is the last element in the namespaced device kind. |
+| *value* | The value of the reading. |
+| *timestamp* | An RFC3339 timestamp describing the time at which the reading was taken. |
+| *unit* | The unit of measure for the reading. If there is no unit, this will be `null`. |
+| *context* | A mapping of arbitrary values to provide additional context for the reading. |
+
+
+##### Error (500, 400)
+See: [Errors](#errors)
+
+
+---
+
+### Write
+```
+POST http://HOST:5000/synse/v3/write/{device_id}
+```
+
+Write data to a device.
+
+Not all devices support writing. The [`/info`](#info) endpoint specifies whether the device
+supports writing under the *capabilities* field. 
+
+The write behavior is defined at the plugin level, so devices with of the same type but
+managed by different plugins may require different inputs to complete.
+
+If the specified device does not support writing, or the provided device ID does not match
+any known device, an error is returned.
+
+Validation of the POSTed write data is done at the plugin level. An unsupported write action
+or invalid write data will result in an error.
+
+For more details on the changes to the `/write` endpoint, see the 
+[Synse Writes Document](./writes.md)
+
+#### URI Parameters
+
+| Parameter | Required | Description |
+| --------- | -------- | ----------- |
+| *device_id* | yes | The ID of the device that is being written to. |
+
+#### Query Parameters
+
+| Key | Description |
+| --- | ----------- |
+| transaction | A user-defined transaction ID which will be used as the transaction id for the write. If this conflicts with an existing transaction ID, an error is returned. If this is not provided, a transaction ID will be automatically generated for the write action. |
```

Comment:
Having the burden on the client is standard for most asynchronous APIs, I thought. I'm not sure I would want to make the write command always block and get rid of the transaction route, but I would be fine with having something like a `block` or `wait` query parameter on the /write endpoint which would allow the write to block until the transaction resolved.